pkg/apps/strategy/recreate/recreate.go

```diff
@@ -60,7 +61,7 @@ type RecreateDeploymentStrategy struct {
 
 // NewRecreateDeploymentStrategy makes a RecreateDeploymentStrategy backed by
 // a real HookExecutor and client.
-func NewRecreateDeploymentStrategy(client kclientset.Interface, tagClient imageclient.ImageStreamTagsGetter,
+func NewRecreateDeploymentStrategy(client kclientset.Interface, kubeClient kubernetes.Interface, tagClient imageclient.ImageStreamTagsGetter,
```

Comment:
@tnozicka i think it is time to switch this to use external kube client so we don't have to do crazy conversions